From a55868275fbc26758defb0b90a2ae9a79bc90acf Mon Sep 17 00:00:00 2001
From: sgnanase <sundar.gnanasekaran@intel.com>
Date: Wed, 18 Sep 2019 16:27:06 +0530
Subject: [PATCH] Revert "ALSA: hda: Fix 1-minute detection delay when i915
 module is not available"

This reverts commit 72651bbdf3d5e2d587c0a1cc0da977cb608787a5.
---
 sound/hda/hdac_i915.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/sound/hda/hdac_i915.c b/sound/hda/hdac_i915.c
index 3847fe841d33..27eb0270a711 100644
--- a/sound/hda/hdac_i915.c
+++ b/sound/hda/hdac_i915.c
@@ -143,12 +143,10 @@ int snd_hdac_i915_init(struct hdac_bus *bus)
 	if (!acomp)
 		return -ENODEV;
 	if (!acomp->ops) {
-		if (!IS_ENABLED(CONFIG_MODULES) ||
-		    !request_module("i915")) {
-			/* 60s timeout */
-			wait_for_completion_timeout(&bind_complete,
-						   msecs_to_jiffies(60 * 1000));
-		}
+		request_module("i915");
+		/* 60s timeout */
+		wait_for_completion_timeout(&bind_complete,
+					    msecs_to_jiffies(60 * 1000));
 	}
 	if (!acomp->ops) {
 		dev_info(bus->dev, "couldn't bind with audio component\n");
-- 
2.22.0

